### PR TITLE
Milestone 11 : Final Project Review

### PR DESCRIPTION
**Pull request by Nemwel-Boniface**
## Hello Reviewer(s) :wave: :wave: 
As I had made sure that I did most of the PR comments as details in all my previous PRs, I will do a brief summary of them all here:

In the Zoo project I was able to:

- [x] Set up the project including all the required libraries for the project
- [x] Choose an API that would give me numerical data to use
- [x] Set up Redux and my store
- [x] Connect my React components with my Reux store
- [x] Display 10 random animals on each page refresh. Note that the project does not re-render on the switch of the routes
- [x] Create the Details page and made sure that I am displaying the information of the right animal
- [x] Mae tests for both my components as well as the Redux store
- [x] Document my project professionally by making sure that I have a professional README.md file
- [x] Check to make sure that I met all the project requirements
- [x] For this project made sure  to use correct git-flow practices
- [x] On each milestone, I made sure to check and fix all the linter errors

Thank you for taking the time to review my project :pray:

### CODE REVIEW

File: src/components/details.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        setAnimal(an);
+      }
+    });
+  }, []);
+  return (
+    <div className="Details">
+      <button type="button" onClick={() => push('/')}>
+        <i className="fa fa-arrow-left" aria-hidden="true" />
+        Home
+      </button>
```
Comment (TedLivist) [inline src/components/details.js:22]: - [ ] As indicated in the design the **`Back`** button (<) should be placed in the design, the back arrow for the **Details** page should be on the header. Kindly fix that.

  nemwel-back-edit

File: src/components/Zoo.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            minWeight={animal.weight_min}
+            maxWeight={animal.weight_max}
+            lifeSpan={animal.lifespan}
+            habitats={animal.habitat}
+            diet={animal.diet}
+            locattions={animal.geo_range}
+            image={animal.image_link}
+          />
+
+        ))}
```
Comment (TedLivist) [inline src/components/Zoo.js:32]: - [ ] According to the design we have alternating lighter and darker colors for the contents on the homepage. Kindly implement that.

  Your Implementation | Given Design
  :---:|:---:
  nemwel-alternating | jonathas-orig-home

File: src/components/details.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        </p>
+        <p>
+          Diet
+          <span className="right">{animal.diet}</span>
+        </p>
+        <p>
+          Locations
+          <span className="right">{animal.locattions}</span>
+        </p>
+      </div>
```
Comment (TedLivist) [inline src/components/details.js:72]: - [ ] Let's try to follow the design as much as possible. On your Details page, kindly add the stripe color as we have in the template.

- [ ] Ensure you maintain the appropriate spacing between elements.

  Your Implementation | Behance Design
  :---:|:---:
  nemwel-details | jonathas-origin-detail

File: src/components/landing.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+const Landing = () => (
+  <div className="Landing">
+    <img src={lnd} alt="Gif" />
+    <div className="LandingDetails">
+      <h2>Lions</h2>
+      <p>The king of the jungle</p>
+    </div>
+  </div>
+);
+export default Landing;
```
Comment (TedLivist) [inline src/components/landing.js:13]: - [ ] Your landing section on the homepage is not as descriptive of the elements on the homepage as what we have on the design. In the design, **EUROPE** and a map of it describes the content which includes the Czech Republic, Italy and so on which are typically countries in the European continent. In your implementation, **LION** and its image, does not describe the homepage contents. Kindly resolve this.

  Your Implementation | Given Design
  :---:|:---:
  nemwel-landing | akumu-descriptive

File: src/components/Zoo.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { shallowEqual, useSelector } from 'react-redux';
+import Animal from './Animal';
+import Landing from './landing';
+
+const Zoo = () => {
```
Comment (TedLivist) [inline src/components/Zoo.js:6]: - [ ] On your homepage, kindly ensure you implement the filtering mechanism (like a search box) as outlined in the requirements.

  furkan-home

File: src/App.test.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { render, screen } from '@testing-library/react';
-import App from './App';
+import ZooContainer from './ZooContainer';
 
 test('renders learn react link', () => {
-  render(<App />);
+  render(<ZooContainer />);
   const linkElement = screen.getByText(/learn react/i);
   expect(linkElement).toBeInTheDocument();
 });
```
Comment (TedLivist) [inline src/App.test.js:8]: - [ ] One of your tests is failing. Kindly resolve it.

  nemwel-test

File: README.md [hunk 7]
```diff
@@ -1,20 +1,43 @@
+![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# React-Redux Setup
+# Nemwel ZOO
```
Comment (TedLivist) [inline README.md:3]: - [ ] Currently, you have only one unit test for the Zoo component. In your app, kindly add more unit tests for some of your components.

- [ ] **Kindly ensure all implemented changes are pushed to the live demo.**

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Nemwel-Boniface,

Good job so far!
✔️ Good documentation
✔️ Connection to the store using Redux hooks
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/details.js (hunk 3 above)
Comment (Nemwel-Boniface) [reply src/components/details.js:72]: Hello @TedLivist Thank you I have effected the change.

File: src/components/details.js (hunk 1 above)
Comment (Nemwel-Boniface) [reply src/components/details.js:22]: Hello @TedLivist thank you for this review.
I have effected this change

File: src/components/Zoo.js (hunk 5 above)
Comment (Nemwel-Boniface) [reply src/components/Zoo.js:6]: Hello @TedLivist I have effected this requested additional feature

**Review by Meltrust — APPROVED** [review]
Hi @Nemwel-Boniface,


## Wow, you did it 🎉
## Congratulations on finishing React!!!
Brilliant
## Thank you for adding the back button, implementing the darker color, implement correct spacing  from the elements, making the landing section more descriptive, adding a filtering mechanism, and adding more tests, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**